test/yaml/akri-debug-echo-foo.yaml

```diff
@@ -4,11 +4,13 @@ metadata:
   name: akri-debug-echo-foo
 spec:
   protocol: 
-    debugEcho:
-      descriptions:
-      - "foo0"
-      - "foo1"
-      shared: false
+    name: debugEcho
+    discoveryDetails:
+      protocolHandler: |+
+        debugEcho:
```

Comment:
do we need the protocol name in 2 places (seems redundant to specify `name: debugEcho` and `protocolHandler: debugEcho: ...`?

Reply:
I wanted to remove the repeat of the protocol name in `protocolHandler` but couldnt find a way to [match for embedded discovery handlers](https://github.com/kate-goldenring/akri/blob/4fe69191245dc1b14c76ef2b70f3150a340156a8/agent/src/util/embedded_discovery_handlers.rs#L51) without the union-like enum.

Reply:
are we requiring that the value be the same?  if we get `name: foo` with `protocolHandler: bar: ...`, do we ignore the settings in `bar` or is it up to the DH?  do we want to enforce a convention somehow (only pass to the DH what what corresponds to `name: XXX`)

Reply:
fwiw, couldn't you pass the value in `name: XXX` into inner_get_discovery_handler to do the match (against the value stored in each DH's PROTOCOL_NAME)?

Reply:
fixed in 0c51141. Removed the enum and instead am matching for protocol_names in `embedded_discovery_handlers.rs`.